proxy/src/telemetry/metrics/latency.rs

```diff
@@ -0,0 +1,285 @@
+#![deny(missing_docs)]
+use control::pb::telemetry::{
+    Latency as LatencyProto,
+};
+
+use std::{ops, slice, u32};
+use std::cell::Cell;
+use std::default::Default;
+use std::time::Duration;
+
+/// A series of latency values and counts.
+#[derive(Debug)]
+pub struct Latencies(Buckets<Latency, Buckets<Latency>>);
+
+/// A latency in tenths of a millisecond.
+#[derive(Debug, Default, Eq, PartialEq, Ord, PartialOrd, Copy, Clone, Hash)]
+pub struct Latency(u32);
+
+/// Generalization of a histogram bucket counting observations of type `T`.
+trait Bucket<T: PartialOrd> {
+    /// Returns `true` if a given `observation` falls into this `Bucket`.
+    fn contains(&self, observation: &T) -> bool;
+    /// Observe a value, incrementing this bucket if it `contains` that value.
+    /// 
+    /// # Returns
+    /// - `true` if this `Bucket`'s count was incremented;
+    /// - `false` otherwise.
+    fn add(&self, observation: &T) -> bool;
```

Comment:
This should probably be `&mut self`, since it is definitely trying to mutate the bucket.